File: Saams.Net/Saams.EF/LeaveManagement/LeaveCounter.cs
using Saams.EF.UserManagement;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Saams.EF.LeaveManagement
{
    [Table("LeaveCounters")]
    public class LeaveCounter : IEntity
    {
        public int Id { get; set; }

        [Required]
        public required int Counter {  get; set; }

        [Required]
        public DateOnly Expiry {  get; set; }

        [Required]
        public required int UserId { get; set; }
        public User User { get; set; } = null!;

        [Required]
        public required int LeaveId { get; set; }
        public Leave Leave { get; set; } = null!;
    }
}

File: saams-ui/src/model/leave-counter.ts
import { Base } from "./base";
import Response from "./response";

export interface LeaveCounter extends Base {
    counter: number;
    expiry: string; // D.HH:mm:ss
    userId: number;
    leaveId: number;
}

export interface LeaveCounterResponse extends Response {
    leaveCounters: LeaveCounter[];
    leaveCounter: LeaveCounter;
}